pkg/reconciler/pipeline/dag/dag_test.go

```diff
@@ -669,3 +691,56 @@ func TestBuild_ConditionsParamsFromTaskResults(t *testing.T) {
 	}
 	assertSameDAG(t, expectedDAG, g)
 }
+
+// This test make sure we detect cycle when it exists.
+// This means we "visit" "a" twice, from two different paths.
```

Comment:
I think this sentence " This means we "visit" "a" twice, from two different paths." is making this more confusing than it needs to be - the problem is that a needs to run after e but e needs to run after a (details of the implementation around "visiting" i dont think really help that much)

Reply:
I have rephrased the comments.